repository.go

```diff
@@ -534,6 +529,43 @@ func (r repository) update(cw contextWrapper, doc *Document, mutation Mutation,
 	return nil
 }
 
+func (r repository) applyMutates(cw contextWrapper, doc *Document, mutation Mutation, filter FilterQuery) error {
+	var (
+		baseQueries = []Querier{filter, mutation.Unscoped, mutation.Cascade}
+		queries     = baseQueries
+	)
+	if version, ok := r.lockVersion(*doc, mutation.Unscoped); ok {
+		Inc("lock_version").SkipReload().Apply(doc, &mutation)
+		queries = append(queries, LockVersion(version))
```

Comment:
And add a query + mutation if the document has a version lock and the query is scoped